mesonbuild/backend/ninjabackend.py

```diff
@@ -1091,6 +1095,42 @@ def generate_vala_compile(self, target, outfile):
         element.write(outfile)
         return other_src[0], other_src[1], vala_c_src
 
+    def generate_nim_target(self, target, outfile):
+        nim = target.compilers['nim']
+        relsrc = []
+        main_nim_file = None
+        for i in target.get_sources():
+            if not nim.can_compile(i):
+                raise InvalidArguments('Nim target %s contains a non-nim source file.' % target.get_basename())
+            if main_nim_file is None:
+                main_nim_file = i.rel_to_builddir(self.build_to_src)
+            relsrc.append(i.rel_to_builddir(self.build_to_src))
+        target_name = os.path.join(target.subdir, target.get_filename())
+        # nim always outputs dep files to the name of the entry point nim file
+        main_file_name = os.path.splitext(main_nim_file.fname)[1]
+        nimcache_dir = nim.get_nimcache_dir(self.get_target_private_dir(target))
+        main_json_file = os.path.join(nimcache_dir, main_file_name + '.json')
```

Comment:
__[Flake8]__

[F841] local variable 'main_json_file' is assigned to but never used